testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py

```diff
@@ -3215,6 +3215,260 @@ def test_replication_push_replication_guest_enabled(params_from_base_test_setup,
+def test_doc_removal_with_multipleChannels(params_from_base_test_setup, setup_customized_teardown_test):
+    """
+        @summary:
+        1. Create users in SGW with multiple channels
+            user A -> channelA,channelB, channel C;
+            userB -> channelB,
+            userC-> channelC
+        2. create docs in SGW
+            doc a with channel A, channelB ;
+            docb with channel B ,
+            docc with ChannelA, channelB, channelC
+        3. Verify User A can access docA and docC.
+            docB by UserB, UserA
+            docC by user A, user C
+        4. Remove the channel c from all the docs.
+        5. Verify userA can access only docA and doc B, but not docC
+            UserB can access docB
+            UserC cannot access docC
+    """
+
+    sg_db = "db"
+    sg_url = params_from_base_test_setup["sg_url"]
+    sg_admin_url = params_from_base_test_setup["sg_admin_url"]
+    sg_blip_url = params_from_base_test_setup["target_url"]
+    base_url = params_from_base_test_setup["base_url"]
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    sg_config = params_from_base_test_setup["sg_config"]
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    sg_config = params_from_base_test_setup["sg_config"]
+    db = params_from_base_test_setup["db"]
+    cbl_db1 = setup_customized_teardown_test["cbl_db1"]
+    cbl_db2 = setup_customized_teardown_test["cbl_db2"]
+    cbl_db3 = setup_customized_teardown_test["cbl_db3"]
+
+    sync_gateway_version = params_from_base_test_setup["sync_gateway_version"]
+
+    username_A = "autotestA"
+    username_B = "autotestB"
+    username_C = "autotestC"
+    password = "password"
+    num_of_docs = 1
+
+    if sync_gateway_version < "2.5.0":
+        pytest.skip('This test cannnot run with sg version below 2.5.0')
+
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+
+    channel_A = ["ABC", "DEF", "XYZ"]
+    channel_B = ["DEF"]
+    channel_C = ["XYZ"]
+
+    doc_channel_1 = ["ABC", "DEF"]
+    doc_channel_2 = ["DEF"]
+
+    sg_client = MobileRestClient()
+    authenticator = Authenticator(base_url)
+    replicator = Replication(base_url)
+
+    # 1. Create users in SGW with multiple channels
+    sg_client.create_user(sg_admin_url, sg_db, username_A, password=password, channels=channel_A)
+    cookie_A, session_id_A = sg_client.create_session(sg_admin_url, sg_db, username_A)
+    session_A = cookie_A, session_id_A
+
+    sg_client.create_user(sg_admin_url, sg_db, username_B, password=password, channels=channel_B)
+    cookie_B, session_id_B = sg_client.create_session(sg_admin_url, sg_db, username_B)
+    session_B = cookie_B, session_id_B
+
+    sg_client.create_user(sg_admin_url, sg_db, username_C, password=password, channels=channel_C)
+    cookie_C, session_id_C = sg_client.create_session(sg_admin_url, sg_db, username_C)
+    session_C = cookie_C, session_id_C
+
+    """ 2. create docs in SGW
```

Comment:
triple quote as specifically intend for doc string. Use # for each line. Also, can you make summary consistent for variable name.

Reply:
done